apis/Google.Cloud.Compute.V1/Google.Cloud.Compute.V1.EnumConstantGenerator/EnumContainer.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Google.Cloud.Compute.V1.EnumConstantGenerator
+{
+    /// <summary>
+    /// A container for an either enums or other nested enum containers.
+    /// </summary>
+    public class EnumContainer
+    {
+        public MessageDescriptor Descriptor { get; }
+
+        public List<EnumContainer> NestedContainers { get; }
+
+        public List<EnumDescriptor> Enums { get; }
+
+        public bool ShouldGenerate => Enums.Count > 0 || NestedContainers.Any(nc => nc.ShouldGenerate);
+
+        public EnumContainer(MessageDescriptor descriptor, IEnumerable<EnumDescriptor> enums, IEnumerable<MessageDescriptor> messages)
+        {
+            Descriptor = descriptor;
+            Enums = enums.ToList();
+            NestedContainers = messages.Select(message => new EnumContainer(message, message.EnumTypes, message.NestedTypes)).ToList();
+        }
+
+        public void Generate(TextWriter writer, string indentation)
+        {
+            if (Descriptor is object)
+            {
+                writer.WriteLine($"{indentation}/// <summary>Container class for enums within the <see cref=\"global::{Descriptor.ClrType.FullName.Replace("+", ".")}\"/> message.</summary>");
+                writer.WriteLine($"{indentation}public static class {Descriptor.ClrType.Name}");
+            }
+            else
+            {
+                // Root class
+                writer.WriteLine($"{indentation}/// <summary>Helper constants with the wire representation for enums within the Compute API.</summary>");
+                writer.WriteLine($"{indentation}public static class ComputeEnumConstants");
+            }
+            writer.WriteLine($"{indentation}{{");
+            writer.WriteLine();
+            string moreIndentation = indentation + "    ";
+            foreach (var enumDescriptor in Enums.OrderBy(d => d.ClrType.Name, StringComparer.Ordinal))
+            {
+                GenerateEnumConstants(writer, enumDescriptor, moreIndentation);
+            }
+            foreach (var container in NestedContainers.Where(nc => nc.ShouldGenerate).OrderBy(nc => nc.Descriptor.Name, StringComparer.Ordinal))
+            {
+                container.Generate(writer, moreIndentation);
+            }
+            writer.WriteLine($"{indentation}}}");
+            writer.WriteLine();
+        }
+
+        private static void GenerateEnumConstants(TextWriter writer, EnumDescriptor descriptor, string indentation)
+        {
+            var enumType = descriptor.ClrType;
+            writer.WriteLine($"{indentation}/// <summary>Constants for wire representations of the <see cref=\"global::{enumType.FullName.Replace("+", ".")}\"/> enum.</summary>");
+            writer.WriteLine($"{indentation}public static class {descriptor.ClrType.Name}");
+            writer.WriteLine($"{indentation}{{");
+            writer.WriteLine();
```

Comment:
nit: I think I'd remove this one.

Reply:
Have changed it to only add a blank line where necessary, both within containers and enums themselves.